src/rng/src/main.rs

```diff
@@ -0,0 +1,106 @@
+// VIRTIO RNG Emulation via vhost-user
+//
+// Copyright 2021 Linaro Ltd. All Rights Reserved.
+//          Mathieu Poirier <mathieu.poirier@linaro.org>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+mod vhu_rng;
+
+use clap::{load_yaml, App};
+
+use std::sync::{Arc, RwLock};
+use std::thread;
+use vhost::vhost_user::Listener;
+use vhost_user_backend::VhostUserDaemon;
+use vhu_rng::VuRngBackend;
+
+fn start_backend(cmd_args: clap::ArgMatches, dry_run: bool) -> Result<(), String> {
+    let mut handles = Vec::new();
+
+    let path = cmd_args.value_of("socket_path").ok_or("Invalid socket path")?;
+
+    let count = match cmd_args.value_of("socket_count") {
+        Some(count) => count.parse::<u32>().unwrap(),
+        None => 1,
+    };
+
+    let source = match cmd_args.value_of("rng_source") {
+        Some(source) => source.to_string(),
+        None => "/dev/urandom".to_string(),
```

Comment:
We will end up using the same filepath for all the guests ? Would the guests ever want separate devices ?

Reply:
Did you actually try to run two guests with the same daemon then ? I wonder if it will actually work. The problem is that you are creating separate backends for each guest, where each backend will open the file /dev/urandom (or whatever). Perhaps only one of them will pass and others will just fail ? You better share the opened file between all the instances then. I did something like that in case of i2c.

Reply:
Indeed I did tested this and /dev/urandom can accomodate multiple readers.

This is something I grappled with during implementation.  If a source file is shared between guests latency caused by contention on a mutex may become a problem.  On the flip side your comment made me think that not all drivers will be implemented the same ways as /dev/urandom and it is very possible some will only support a single reader.   

Contention, if it becomes a problem, can be dealt with by spinning of multiple instances of the daemon and let the VFS layer deal with concurrent accesses.  But if a driver only supports a single reader than the current implementation can only serve a single guest, which is quite limiting.  As such I will proceed the same way you did for I2c.